### DISCUSSION TITLE
Backup api

### DISCUSSION THREAD

**Discussion by AnupNadahalli** [xerial/sqlite-jdbc#775] [Q&A] [answered]
Hi All,
Currently I am looking at backup api present in org.sqlite.core.DB class which is abstract and is implemented by org.sqlite.core.NativeDB class. I am able to backup my in memory database to disk file using this api. But I wanted to know if it is following the same approach as the one mentioned in sqlite official documentation (https://sqlite.com/backup.html).

**Comment by gotson** [comment, marked as answer]
> backup api present in org.sqlite.SQLiteConnection class

i don't see any API of that kind in that class, would you be able to precise what you are referring to?

**Reply by AnupNadahalli** [reply]
Sorry. My mistake. I am talking about org.sqlite.core.DB.backup method. I have corrected it.

**Reply by gotson** [reply]
You can see what that does [here](https://github.com/xerial/sqlite-jdbc/blob/4102c1d445c2de550a62a2281c44bd3c86cfc139/src/main/java/org/sqlite/core/NativeDB.c#L1467-L1487)

**Reply by AnupNadahalli** [reply]
Thanks a lot. It looks like it is doing the same thing as mentioned in sqlite online backup tool.
